apiserver/src/api/mod.rs

```diff
@@ -101,6 +104,9 @@ pub async fn run_server<T: 'static + BottlerocketShadowClient>(
     k8s_client: kube::Client,
     prometheus_exporter: Option<opentelemetry_prometheus::PrometheusExporter>,
 ) -> Result<()> {
+    let public_key_path = format!("{}/{}", TLS_KEY_MOUNT_PATH, PUBLIC_KEY_NAME);
+    let certificate_cache = read_certificate(&public_key_path).unwrap();
```

Comment:
Let's replace this `unwrap` with a structured error in the API module (reference: https://github.com/bottlerocket-os/bottlerocket/commit/5a58d4b2b98dbed06a103d866b4b753f580f0e32)